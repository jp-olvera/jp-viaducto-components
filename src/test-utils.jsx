import React from 'react';
import { render } from '@testing-library/react';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }) => {
  return <ConfigProvider>{children}</ConfigProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
