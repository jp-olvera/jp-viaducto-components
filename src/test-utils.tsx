import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }: any) => (
  <ConfigProvider>{children}</ConfigProvider>
);

jest.mock('./cells/Input/amex.svg', () => null);
jest.mock('./cells/Input/mastercard.svg', () => null);
jest.mock('./cells/Input/visa.svg', () => null);
jest.mock('./cells/Dropdown/sorting.svg', () => null);
jest.mock('./cells/BareButton/close.svg', () => null);

const customRender = (ui, options?) => render(ui, { wrapper: AllTheProviders, ...options });
// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
