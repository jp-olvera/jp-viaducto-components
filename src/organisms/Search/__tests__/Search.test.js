/* eslint-env jest */

import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import '@testing-library/react';
import { render, screen, fireEvent } from '../../../test-utils';
import { Search } from '..';

jest.mock('../../../cells/Dropdown/sorting.svg', () => null);

const options = ['Raz√≥n Social', 'RFC', 'Nombre Comercial'];
describe('<Search/>', () => {
  test('should render properly', () => {
    const handleClick = jest.fn();
    render(
      <Search
        buttonLabel='Search 1'
        handleSearch={handleClick}
        options={options}
        family='Roboto'
        id='search'
      />,
    );
    expect(screen.getByTestId('search')).toBeVisible();
  });
  test('should render properly with no options', () => {
    const handleClick = jest.fn();
    render(
      <Search
        buttonLabel='Search 1'
        handleSearch={handleClick}
        id='searchNoOptions'
      />,
    );
    expect(screen.getByTestId('searchNoOptions')).toMatchSnapshot();
  });

  test('call function on button press', () => {
    const handleClick = jest.fn().mockReturnValue(() => null);
    render(
      <Search
        buttonLabel='Search 2'
        handleSearch={handleClick}
        options={options}
        id='search'
      />,
    );
    fireEvent.click(screen.getByTestId('button-search'));
    expect(handleClick).toBeCalled();
  });

  test('should write in input search', () => {
    const value = 'Search this value';
    const { getByTestId } = render(
      <Search buttonLabel='Search 3' options={options} id='search' />,
    );
    const input = getByTestId('input-search');
    fireEvent.change(input, { target: { value } });
    expect(input.value).toContain(value);
  });

  test('should display dropdown', () => {
    const handleClick = jest.fn();
    const { getByTestId } = render(
      <Search
        buttonLabel='Search 4'
        options={options}
        id='search'
        handleSearch={handleClick}
      />,
    );
    const activator = getByTestId('select');
    activator.click();
    expect(screen.getByText('RFC')).toBeVisible();
  });

  test('should display dropdown with no props', () => {
    const { queryByText, getByTestId } = render(
      <Search family={null} options={['option1', 'option2']} />,
    );
    const activator = getByTestId('select');
    activator.click();
    expect(queryByText('option1')).toBeInTheDocument();
    expect(queryByText('option1')).toBeInTheDocument();
  });
});
