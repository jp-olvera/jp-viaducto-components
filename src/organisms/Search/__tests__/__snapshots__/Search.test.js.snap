// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search/> should render properly with no options 1`] = `
<div
  class="StyledSearch-ku5ztu-0 etPTiU"
  data-testid="searchNoOptions"
  elevation="1"
>
  <input
    data-testid="input-searchNoOptions"
    id="searchNoOptions"
    placeholder="Introduce el término de búsqueda"
    type="text"
  />
  <div
    class="StyledSelect__StyledSelectWrapper-tky0tx-0 hFmgKk"
  >
    <select
      class="StyledSelect-tky0tx-1 gTuCNZ"
      data-testid="select"
    />
  </div>
  <button
    class="StyledButton-hwi8uc-0 iBkvkG"
    data-testid="button-searchNoOptions"
    height="2.488rem"
    type="button"
  >
    <span
      class="button-icon-span"
    >
      <svg
        aria-labelledby="searchIconTitle"
        color="#FFFFFF"
        fill="none"
        height="1em"
        role="img"
        stroke="currentColor"
        stroke-linecap="square"
        stroke-linejoin="miter"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
      >
        <title
          id="searchIconTitle"
        >
          Search
        </title>
        <path
          d="M14.4121122,14.4121122 L20,20"
        />
        <circle
          cx="10"
          cy="10"
          r="6"
        />
      </svg>
    </span>
    <span>
      Search 1
    </span>
  </button>
</div>
`;
